bip-schnorr.mediawiki

```diff
@@ -76,6 +77,8 @@ expensive conversion to affine coordinates first. This would even be the case if
 
 '''Final scheme''' As a result, our final scheme ends up using signatures ''(r,s)'' where ''r'' is the X coordinate of a point ''R'' on the curve whose Y coordinate is a quadratic residue, and which satisfies ''sG = R + H(r || P || m)P''.
 
+'''Optional smaller encoding of the public key''' In order to allow encoding the public key as 32 bytes instead of 33 bytes, our scheme optionally allows signing for the public key with even Y coordinate. In particular, for a given secret key the signing algorithm checks if the corresponding public key has an even Y coordinate. If that's not the case, secret and public key are negated (which results in a public key with an even Y coordinate). The signer can then send only the last 32 bytes of the compressed public key along with the signature to the verifier. After receiving the 32 byte public key part, the verifier adds the byte ''0x02'' to the beginning which allows verification of the signature.
```

Comment:
Perhaps worth pointing out that there is no reduction in security (when expressed as the number of EC operations an attacker is expected to perform to break the DLP), as point negation can be done without EC operations.

Reply:
added

Reply:
Nice, your paragraph describes the basic idea of the reduction that shows that the 32 byte-variant is as secure as the normal variant. I think this can be expanded slightly, I can suggest some sentences later.